### ISSUE TITLE
charset=utf-8 does not warn about ISO-8859-1 file

### ISSUE THREAD

**Issue by Luflosi** [editorconfig-checker/editorconfig-checker#597]
`editorconfig-checker` does not complain that a file is not encoded with UTF-8 even though `charset = utf-8` is set.

.editorconfig:
```.editorconfig
root = true

[*]
charset = utf-8
```

Create a text file encoded with ISO-8859-1:
```sh
printf 'some\xa0text' > test.txt
```
Hex a0 is a non-breaking space in ISO-8859-1. As far as I can tell, this file is not valid UTF-8.

Check that the contents are correct with `xxd test.txt`:
```
00000000: 736f 6d65 a074 6578 74                   some.text
```

`file --mime test.txt`:
```
test.txt: text/plain; charset=iso-8859-1
```

If I change `charset = utf-8` to `charset = ascii`, it complains as expected:
```
test.txt:
	Wrong character encoding ("ISO-8859-1" instead of "ascii")

1 errors found
```

**Comment by Luflosi** [comment]
I'm using the latest version 3.11.1 BTW.

**Comment by Luflosi** [comment]
Thank you very much for fixing this so quickly!
